components/scream/src/physics/shoc/tests/shoc_diag_second_mom_lbycond_test.cpp

```diff
@@ -0,0 +1,92 @@
+#include "catch2/catch.hpp"
+
+#include "shoc_unit_tests_common.hpp"
+
+#include "physics/shoc/shoc_functions.hpp"
+#include "physics/shoc/shoc_functions_f90.hpp"
+#include "physics/share/physics_constants.hpp"
+#include "share/scream_types.hpp"
+
+#include "ekat/ekat_pack.hpp"
+#include "ekat/util/ekat_arch.hpp"
+#include "ekat/kokkos/ekat_kokkos_utils.hpp"
+
+#include <algorithm>
+#include <array>
+#include <random>
+#include <thread>
+
+namespace scream {
+namespace shoc {
+namespace unit_test {
+
+template <typename D>
+struct UnitWrap::UnitTest<D>::TestSecondMomLbycond {
+
+static void run_second_mom_lbycond_bfb()
+{
+  SHOCSecondMomLbycondData lby_fortran[] = {
+    // shcol
+    SHOCSecondMomLbycondData(120),
+    SHOCSecondMomLbycondData(120),
+    SHOCSecondMomLbycondData(120),
+    SHOCSecondMomLbycondData(120),
+  };
+
+  static constexpr Int num_runs = sizeof(lby_fortran) / sizeof(SHOCSecondMomLbycondData);
+
```

Comment:
Need to generate some random input data here (randomize method on the Data objects), otherwise we are just testing against all-zero input, which is why the bfb problems were not caught.

Reply:
This also applies to the test in PR #532 when we introduce non-zero tendencies.

Reply:
yes, add the call to randomize methods